### DISCUSSION TITLE
OAS3 Array query params example

### DISCUSSION THREAD

**Discussion by rafalkrupinski** [OAI/OpenAPI-Specification#3047] [Q&A] [answered]
Hi all

I have some doubts about the examples in 4.7.12.4 - serialization styles.

Field Name | Type | Description
-- | -- | --
explode | boolean | When this is true, parameter values of type array or object generate separate parameters for each value of the array [...]

The input array data for array is 
`color := ["blue","black","brown"]`

I'd expect it to serialize as:
exploded: `'color=blue&color=black&color=brown'`
non-exploded: `'color=blue,black,brown'`

Yet both examples are simply `'blue,black,brown'`

Is this correct?

**Comment by commonism** [comment]
I think this refers to 
https://www.rfc-editor.org/rfc/rfc6570#section-3.2.2
```
       {list}             red,green,blue
       {list*}            red,green,blue
```

**Reply by charjr** [reply, marked as answer]
Yes it does, I asked a similar question before, and got pointed to the same rfc [(You may find it useful to read the answers I received)](https://github.com/OAI/OpenAPI-Specification/issues/3481)

[RFC6570-2.4.2](https://www.rfc-editor.org/rfc/rfc6570#section-2.4.2) will give you a good idea of why someone might want explode.

The example you've given is for the **simple** style which can **only be used in paths and headers**:

In a path: parameters MUST appear in the same order specified in the template, so the name isn't required i.e.

`/paint/{colors}/{type}` and you get `/paint/red,green,blue/silk` you still know which part the color parameter is because each parameter is delimited by a forward slash.

In a header: each header has a name, so the name of the parameter is already given.

`color: red,green,blue` is clear.
`color: color=red,green,blue` would be redundant.

**Comment by handrews** [comment]
Answer accepted, closing.
